### ISSUE TITLE
GitHub WebHook ping event: Webhook processing failed: event not defined to be parsed

### ISSUE THREAD

**Issue by GuangTianLi** [argoproj/argo-cd#7555]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

**To Reproduce**

A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue.

```text
Request URL: <WebHook URL>
Request method: POST
Accept: */*
content-type: application/json
User-Agent: GitHub-Hookshot/04c124e
X-GitHub-Delivery: 33f01390-36cd-11ec-94f4-a0896a480161
X-GitHub-Event: ping
X-GitHub-Hook-ID: 325405283
X-GitHub-Hook-Installation-Target-ID: 421364558
X-GitHub-Hook-Installation-Target-Type: repository
X-Hub-Signature: sha1=f92aec6a8e42100292bab7d3a155d0dfd91c63b4
X-Hub-Signature-256: sha256=592c7264f830738bde2fe6eec8341b2e343183608dc0e44c84311959de459343
```

**Expected behavior**

`Response 200`

**Screenshots**

**Version**

```shell
Paste the output from `argocd version` here.
argocd: v2.1.3+d855831
  BuildDate: 2021-09-29T21:51:21Z
  GitCommit: d855831540e51d8a90b1006d2eb9f49ab1b088af
  GitTreeState: clean
  GoVersion: go1.16.5
  Compiler: gc
  Platform: linux/amd64
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by BrianHicks** [comment]
I noticed this happening too! To save future issue searchers some work: Argo CD 2.1.5 fails on the `ping` event, but works on a `push` event.

You can tell which was which by:

1. going to the GitHub UI
2. opening your webhook
3. opening the delivery history
4. opening an event
5. examining the `X-GitHub-Event` header

**Comment by tarrychk** [comment]
This is similar to #6029 which was fixed by #6034.
https://github.com/argoproj/argo-cd/blob/e01ab05d555b799cba08097710482afac3be601e/util/webhook/webhook.go#L388 needs to be changed to `payload, err = a.github.Parse(r, github.PushEvent, github.PingEvent)`.
